vllm_v1/worker/gpu_model_runner.py

```diff
@@ -0,0 +1,704 @@
+                self.persistent_batch.token_ids_cpu[i, seq_len] = token_id
+                req_state.output_token_ids.append(token_id)
+            else:
+                # Ignore the sampled token from the partial request.
+                # Rewind the generator state as if the token was not sampled.
+                generator = self.persistent_batch.generators[i]
+                if generator is not None:
+                    offset = generator.get_offset()
+                    generator = generator.set_offset(offset - 1)
+                    self.persistent_batch.generators[i] = generator
+
+        if sampler_output.logprob_token_ids is None:
+            logprob_token_ids = None
+        else:
+            logprob_token_ids = sampler_output.logprob_token_ids.cpu()
+        if sampler_output.logprobs is None:
+            logprobs = None
+        else:
+            logprobs = sampler_output.logprobs.cpu()
+        model_runner_output = ModelRunnerOutput(
+            req_ids=self.persistent_batch.req_ids[:num_reqs],
+            req_id_to_index=self.persistent_batch.req_id_to_index,
+            sampled_token_ids_cpu=sampled_token_ids,
+            logprob_token_ids_cpu=logprob_token_ids,
+            logprobs_cpu=logprobs,
+        )
+        return model_runner_output
+
+    def load_model(self) -> None:
+        logger.info("Starting to load model %s...", self.model_config.model)
+        from unittest.mock import patch
+        from vllm_v1.sample.logits_processor import LogitsProcessor
+        from vllm_v1.sample.sampler import Sampler
+        with DeviceMemoryProfiler() as m:
+            with patch(
+                    "vllm.model_executor.layers.logits_processor.LogitsProcessor",
+                    LogitsProcessor):
+                with patch("vllm.model_executor.layers.sampler.Sampler",
+                           Sampler):
+                    self.model = get_model(
+                        model_config=self.model_config,
+                        device_config=self.device_config,
+                        load_config=self.load_config,
+                        lora_config=self.lora_config,
+                        parallel_config=self.parallel_config,
+                        scheduler_config=self.scheduler_config,
+                        cache_config=self.cache_config)
+
+        self.model_memory_usage = m.consumed_memory
+        logger.info("Loading model weights took %.4f GB",
+                    self.model_memory_usage / float(2**30))
+
+    def _dummy_run(self, model: nn.Module, num_tokens: int) -> None:
+        input_ids = torch.randint(0,
+                                  1000, (num_tokens, ),
+                                  dtype=torch.int32,
+                                  device=self.device)
+        positions = torch.randint(0,
+                                  1000, (num_tokens, ),
+                                  dtype=torch.long,
+                                  device=self.device)
```

Comment:
Nit: the magic number 1000 can lead to error for models with small vocab or short max position embeddings.

Reply:
Good catch. Fixed to `torch.zeros`.